R/kiss-rule.R

```diff
@@ -0,0 +1,102 @@
+#' \code{KissRule.Event} objects are used to define target segments to run a report on.
+#' @param eventId Which event are you targeting on? We need the index or id.
+#' @param frequencyValue How many times the event needs to have happened for the
+#'   rule
+#' @param frequencyOccurance How we are comparing against the
+#'   \code{frequencyValue}. Must be \code{at_least}, \code{at_most}, or
+#'   \code{exactly}
+#' @param comparisonMode - unclear what this does. Defaults to 'any_value'
+#' @param negate Is this an inclusive rule or an exclusionary rule?
+#'
+#' @examples
+#'    firstTimeVisitedCalculation <- KissCalculation.Event(
+#'                      label = "First time of visited site",
+#'                      eventId = 6,
+#'                      type = "first_date_in_range")
+#'    lastTimeVisitedCalculation <- KissCalculation.Event(
+#'                      label = "Last time of visited site",
+#'                      eventId = 6,
+#'                      type = "last_date_in_range")
+#'    reportDates <- lubridate::interval(as.Date("2015-06-01"), as.Date("2015-06-02"))
+#'    rules <- list(KissRule.Event(FALSE, 72, 1, "at_least", "any_value"))
+#'    segment <- KissSegment(type = "and",
+#'                 rules = rules,
+#'                 defaultInterval = reportDates)
+#'    report <- KissReport(productId = "6581c29e-ab13-1030-97f2-22000a91b1a1",
+#'                 segment = segment,
+#'                 calculations = list(
+#'                   firstTimeVisitedCalculation,
+#'                   lastTimeVisitedCalculation
+#'                 ),
+#'                 interval = reportDates
+#'              )
+#'    reportResults <- read(report)
+#' @export
+KissRule.Event <- function(negate, eventId, frequencyValue, frequencyOccurance, comparisonMode = 'any_value') {
```

Comment:
We need to include a date range for each rule - both event and property - the resulting json should look like:
`{
        "type": "event",
        "negate": false,
        "event": 72,
        "frequencyValue": 1,
        "frequencyOccurance": "at_least",
        "comparisonMode": "any_value",
        "dateRange": {
          "dateRangeId": "custom",
          "startDate": "2015-06-01T00:00:00",
          "endDate": "2015-06-02T00:00:00"
        }
      }`